Content.MapRenderer/Painters/GridPainter.cs

```diff
@@ -75,29 +92,63 @@ public void Run(Image gridCanvas, IMapGrid grid)
                         $"No sprite component found on an entity for which a server sprite component exists. Prototype id: {prototype.ID}");
                 }
 
-                var xOffset = 0;
-                var yOffset = 0;
-                var tileSize = 1;
-
                 var transform = _sEntityManager.GetComponent<TransformComponent>(entity);
                 if (_cMapManager.TryGetGrid(transform.GridID, out var grid))
                 {
-                    xOffset = (int) Math.Abs(grid.LocalBounds.Left);
-                    yOffset = (int) Math.Abs(grid.LocalBounds.Bottom);
-                    tileSize = grid.TileSize;
-                }
+                    var position = transform.LocalPosition;
 
-                var position = transform.LocalPosition;
-                var x = ((float) Math.Floor(position.X) + xOffset) * tileSize * TilePainter.TileImageSize;
-                var y = ((float) Math.Floor(position.Y) + yOffset) * tileSize * TilePainter.TileImageSize;
-                var data = new EntityData(sprite, x, y);
+                    var (x, y) = TransformLocalPosition(position, grid);
+                    var data = new EntityData(sprite, x, y);
 
-                components.GetOrAdd(transform.GridID, _ => new List<EntityData>()).Add(data);
+                    components.GetOrAdd(transform.GridID, _ => new List<EntityData>()).Add(data);
+                }
             }
 
             Console.WriteLine($"Found {components.Values.Sum(l => l.Count)} entities on {components.Count} grids in {(int) stopwatch.Elapsed.TotalMilliseconds} ms");
 
             return components;
         }
+
+        private ConcurrentDictionary<GridId, List<DecalData>> GetDecals()
+        {
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+
+            var decals = new ConcurrentDictionary<GridId, List<DecalData>>();
+
+            foreach (var grid in _sMapManager.GetAllGrids())
+            {
+                // TODO this needs to use the client entity manager because the client
+                // actually has the correct z-indices for decals for some reason when the server doesn't,
+                // BUT can't do that yet because the client hasn't actually received everything yet
+                // for some reason decal moment i guess.
```

Comment:
PVS should be off by cvar for the map renderer, so I'm not sure why they wouldn't be sent yet

Reply:
decals dont support net.pvs being off i think isthe issue

Reply:
since they do their own chunking/sending impl